### PR TITLE
WIP: VMSplice/STDOUT consumer

### PR DESCRIPTION
**Pull request by tempusfrangit**

### CODE REVIEW

**Comment by tempusfrangit** [comment]
@technillogue see if this works for you, we chould generally order the requests for slices.

File: .goreleaser.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ builds:
 archives:
   - format: binary
     name_template: >-
-      {{ .ProjectName }}_{{ title .Os }}_
+      {{ .ProjectName }}_{{ tolower .Os }}_
```
Comment (tempusfrangit) [inline .goreleaser.yaml:22]:
```suggestion
      {{ .ProjectName }}_{{ title .Os }}_
```

Comment (tempusfrangit) [reply .goreleaser.yaml:22]: this was not an intended change.

**Comment by technillogue** [comment]
this could definitely work! would it work with multifile and writing multiple files to stdout in order?

**Comment by tempusfrangit** [comment]
We can make that happen! Wouldn't be too hard. This was a first pass to see if it looked good.

**Comment by tempusfrangit** [comment]
It turns out the only reason we don't order downloads which causes you the problem is a historical reason for using a map instead of a list. 

Additionally there is the maximum concurrent files option already that could limit concurrency -- additionally we can set it to automatically use a max file concurrency of one for stdout/vmsplice modes.

**Comment by technillogue** [comment]
how would I invoke this?

**Comment by tempusfrangit** [comment]
@technillogue Rebased. Download order is now guaranteed. You will want to make sure to use `--max-concurrent-files=1` and `-o stdout` or `-o vmsplice`

**Comment by technillogue** [comment]
this doesn't build

```
Error: pkg/consumer/linux_vmsplice.go:15:18: cannot use &VMSpliceConsumer{} (value of type *VMSpliceConsumer) as Consumer value in variable declaration: *VMSpliceConsumer does not implement Consumer (wrong type for method Consume)
		have Consume(io.Reader, string, int64) error
		want Consume(io.Reader, string) error
Error: pkg/consumer/stdout.go:9:18: cannot use &StdoutConsumer{} (value of type *StdoutConsumer) as Consumer value in variable declaration: *StdoutConsumer does not implement Consumer (wrong type for method Consume)
		have Consume(io.Reader, string, int64) error
		want Consume(io.Reader, string) error
```

**Pull request closed without merging**